### PR TITLE
On macOS when launching multiple times focus win

### PR DESCRIPTION
**Pull request by cyphactor**
I dug into this and narrowed the issue down to the macOS app bundle
Info.plist file. So, I spun up a native macOS app real quick and tested
it by launching the binary directly and launching it via the app bundle.
When launching from the command line directly, it created multiple
windows & instances of the app. However, when launching via the app
bundle it behaved as I normally expect a macOS app to behave, that is
when launched multiple times to simply focus the already existing window
and instance.

This informed me that it wasn't something in code as much as it was
something in the app bundle configuration. Hence, I reworked the
Info.plist file based on the one that was created by XCode when I made
the native macOS app and it started behaving as expected.

The downside is that the ALACRITTY_LOCALE, LANG, and LC_CTYPE are not set as
they would be via the launcher script. Also, the PWD starts out at / rather than
the users home directory. The LANG, LC_CTYPE environment variables can be set in
your config under the env section. It isn't dynamic which still isn't quite as
nice as the bash + applescript approach. However, it make the behavior work
correctly which is critical for anyone using an app launcher like Spotlight to
switch apps.

### CODE REVIEW

File: assets/osx/Alacritty.app/Contents/Info.plist [hunk 1]
```diff
@@ -1,32 +1,34 @@
...
-    <key>NSMainNibFile</key>
-    <string></string>
-    <key>NSSupportsAutomaticGraphicsSwitching</key>
-    <true/>
+  <key>CFBundleDevelopmentRegion</key>
+  <string>en</string>
+  <key>CFBundleExecutable</key>
+  <string>alacritty</string>
+  <key>CFBundleIdentifier</key>
+  <string>com.github.jwilm.alacritty</string>
```
Comment (jwilm) [inline assets/osx/Alacritty.app/Contents/Info.plist]: Please make this `io.alacritty`

**Comment by cyphactor** [comment]
@jwilm I made the requested changes, back merged master into my branch, resolved the conflict, and testing things. They seem to be good.

**Comment by jwilm** [comment]
Thanks!

**Comment by cyphactor** [comment]
np

**Comment by unphased** [comment]
So, now, the only way to open more than one window of Alacritty is... `open -n /Applications/Alacritty.app` (assuming I've "installed" it)?

**Comment by cyphactor** [comment]
You can also do `open -a -n ‘Alacritty’`. But, yes, this is the 
intended behavior as this is how macOS apps behave. If an app supports 
multiple windows, it is a feature of the app not that the app is 
actually running multiple times.

**Comment by unphased** [comment]
`open -a -n Alacritty` does not work. it tries to look for the app in the cwd, so `-a` seems superfluous

**Comment by cyphactor** [comment]
Sorry Alacritty needs to follow the -a and I think be in single quotes or double quotes

**Comment by unphased** [comment]
Very interesting. When I do that, it tries to run /Users`/me/Documents/alacritty/assets/osx/Alacritty.app` which fails as it is missing its executable. Seems like lots more can go wrong so I will just use the absolute applications path without `-a` :)

**Comment by cyphactor** [comment]
Sounds like Spotlight has indexed your the Alacritty.app in `/User/me/Documents/alacritty/assets/osx/Alacritty.app` which you don't want.

**Comment by unphased** [comment]
indeed!

spotlight is generally not helpful (eating up your CPU indexing your FS) except for the times that it does turn out to be helpful, so I would never use `open -a` personally, after seeing how it behaves...

my advice would be to use the `open -n /Applications/Alacritty.app` bound to a key to do it, now that using spotlight to spawn Alacritty no longer works. Or, if you don't install Alacritty to Applications, just point it to the absolute path to where it lands in `target/release/osx` after you `make app`.

**Comment by asilvadesigns** [comment]
so... how does one toggle between multiple open instances of Alacritty on MacOS.
`cmd + ~` doesnt work, I'm assuming I have to bind that to something?

**Comment by unphased** [comment]
multiple open instances must be switched as though they're different apps. when alacritty properly supports multiple windows in one instance, then `cmd + ~` will work.

**Comment by unphased** [comment]
This is related to #607, #401, #674. I would like to announce that I have a functioning workaround for `cmd + backtick`. 

You may be able to understand why this works. It's not elegant, but works a damn treat. 

```sh
#!/bin/bash
# open the next program that matches a given process name, if pid of program
# owning the active window is found in the pgrep list, go to the next one in
# the list.
# ME is the pid corresponding to the process owning the focused macos window.
ME=$(osascript -e "tell application \"System Events\" to get unix id of first process whose frontmost is true")
# List of targeted pids
TARGETS=$(pgrep $1 | tr '\n' ' ')
echo $ME -- $TARGETS
TARGETS="$TARGETS$TARGETS"
# construct doubled list to ensure the first instance of ME will not be the
# last. Then focus the very next one.
STRIPFRONT="${TARGETS#*$ME }"
TARGETPID="${STRIPFRONT%% *}"
osascript -e "tell application \"System Events\" to set frontmost of the first process whose unix id is ${TARGETPID} to true"
```

I stash this in my shell scripts dir. You can try it out, it basically will tell OS X/macOS to focus the next alacritty window. One instance of the process corresponds to one window (the way winit works now) Simple as that. My keybinding, as an example: 

```
  - { key: Grave,    mods: Command, command: { program: "/Users/slu/util/macos-open-next-program.sh", args: ["alacritty"] } }
```

**Comment by unphased** [comment]
I'm not sure right now if its something I messed up on my end or not but when I recompiled master this trick in my config file no longer works. Will report back when I get around to fixing this.

**Pull request merged**